biogrid/libs/biogrid-simulator/src/lib/community/ruralarea.ts

```diff
@@ -0,0 +1,56 @@
+import { Town } from '@biogrid/grid-simulator';
+import { Building } from '../building';
+
+/**
+ * A rural area that represents a community where a microgrid can function.
+ */
+export class RuralArea implements Town {
+
+  private town: Building[];
+
+  /**
+   * @param {Building[]} town A list of buildings which make up a town.
+   */
+  constructor(town: Building[]) {
+    this.town = town;
+    this.setCurrentBuildingIds();
+  }
+
+  getTown() : Building[] {
+    return this.town;
+  }
+
+  /**
+   * This method gets a building in a town by its name identifier.
+   * @param {Building} building The building we're looking for.
+   * @return {Building} Returns the building if found in the list, null if not.
+   */
+  getBuildingById(Id: number) {
+    for (let i=0; i<this.town.length; i++) {
+      if (this.town[i].getBuildingId() === Id) {
+        return this.town[i];
+      }
+    }
+    return null;
+  }
+
+  /**
+   * This method adds a building to our town.
+   * @param {Building} newBuilding The building to be added. 
+   */
+  addBuilding(newBuilding: Building) : Building {
+    newBuilding.setBuildingId(this.town.length);
```

Comment:
This is a slightly dangerous way to generate IDs - what happens if you have 4 buildings with the following IDs: 
0 1 2 3 
And then you remove building with ID 1, now you have buildings 0, 2, and 3 
Trying to add a new building, your town.length will be 3, so the new building ID will be 3. A different building with ID 3 already exists, and now you are going to have a duplicate building ID 
You don't currently have the functionality to remove a building, but I think this is still important to account for. I would try something like generating a random number as an ID, looping through your existing buildings to make sure  none of them have this ID already, and then setting this random number as the building ID

Reply:
sounds good will add that.